configure.ac

```diff
@@ -520,8 +520,7 @@ AC_ARG_ENABLE([econf],
   AS_HELP_STRING([--disable-econf], [do not use libeconf]),
   [WITH_ECONF=$enableval], WITH_ECONF=yes)
 if test "$WITH_ECONF" = "yes" ; then
-  PKG_CHECK_MODULES([ECONF], [libeconf], [],
-  [AC_CHECK_LIB([econf],[econf_readDirs],[ECONF_LIBS="-leconf"],[ECONF_LIBS=""])])
+  PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0], [ECONF_LIBS="-leconf"], [ECONF_LIBS=""] )
```

Comment:
I wonder wouldn't a simpler variant be enough, e.g.
```
PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0])
```

Reply:
Yes, that should be done in a more simple way. Fixed.
There is another PR which checks this too. I will fix it too.

Reply:
Hm, the "problem" is the buid PAM environment. It does not contain libeconf but configure will be called without the option --disable-econf. So it is is so "complicate" because libeconf will be disabled for the build if it is not there. So I have reverted it in order to get a running build again:
+++ b/configure.ac
@@ -520,9 +520,10 @@ AC_ARG_ENABLE([econf],
   AS_HELP_STRING([--disable-econf], [do not use libeconf]),
   [WITH_ECONF=$enableval], WITH_ECONF=yes)
 if test "$WITH_ECONF" = "yes" ; then
-  PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0])
-  ECONF_LIBS="-leconf"
-  ECONF_CFLAGS="-DUSE_ECONF=1 $ECONF_CFLAGS"
+  PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0], [ECONF_LIBS="-leconf"],[ECONF_LIBS=""])
+  if test -n "$ECONF_LIBS" ; then
+    ECONF_CFLAGS="-DUSE_ECONF=1 $ECONF_CFLAGS"
+  fi
 fi

Either we put libeconf in the PAM build environment, or we disabling libeconf via --disable-econf in the configure calls or we disable libeconf (almost) silently if it is not there (current state in the PR again).
So, what is would be prefered ?

Reply:
> Sorry, I don't understand what was the problem with `PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0])`.

With your suggestion, configure will fail if libeconf is not installed and you can only build Linux-PAM if you use --disable-econf.
Because this statement makes libeconf >= 0.5.0 a hard requirement.
With Stefans solution, the test will not fail, but libeconf support get's disabled if libeconf is not installed.

Reply:
I see.  I suppose this would work:
```
ECONF_CFLAGS=
ECONF_LIBS=
AC_ARG_ENABLE([econf],
  AS_HELP_STRING([--disable-econf], [do not use libeconf]),
  [WITH_ECONF=$enableval], [WITH_ECONF=yes])
if test "$WITH_ECONF" = "yes"; then
  PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0], [], [])
  if test -n "$ECONF_LIBS"; then
    ECONF_CFLAGS="-DUSE_ECONF=1 $ECONF_CFLAGS"
  fi
fi
AC_SUBST([ECONF_CFLAGS])
AC_SUBST([ECONF_LIBS])
```

By the way, it should have been called --with-econf/--without-econf rather than --enable-econf/--disable-econf.

Reply:
Or even a shorter form:
```
ECONF_CFLAGS=
ECONF_LIBS=
AC_ARG_ENABLE([econf],
  AS_HELP_STRING([--disable-econf], [do not use libeconf]),
  [WITH_ECONF=$enableval], [WITH_ECONF=yes])
if test "$WITH_ECONF" = "yes"; then
  PKG_CHECK_MODULES([ECONF], [libeconf >= 0.5.0], [ECONF_CFLAGS="-DUSE_ECONF=1 $ECONF_CFLAGS"], [:])
fi
AC_SUBST([ECONF_CFLAGS])
AC_SUBST([ECONF_LIBS])
```

Reply:
Thank you !!!!